File: test/src/main/java/com/tunjid/androidbootstrap/test/idlingresources/FragmentVisibleIdlingResource.java
package com.tunjid.androidbootstrap.test.idlingresources;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v7.app.AppCompatActivity;

/**
 * Idling resource that waits for a fragment to be visible.
 * <p>
 * Created by tj.dahunsi on 4/29/17.
 */
public class FragmentVisibleIdlingResource extends BaseFragmentIdlingResource {
    public FragmentVisibleIdlingResource(AppCompatActivity activity, String fragmentTag, boolean unregisterSelf) {
        super(activity, fragmentTag, unregisterSelf);
    }

    public FragmentVisibleIdlingResource(FragmentManager fragmentManager, String fragmentTag, boolean unregisterSelf) {
        super(fragmentManager, fragmentTag, unregisterSelf);
    }

    @Override
    public boolean isIdleNow() {
        Fragment fragment = findFragmentByTag();

        // Wait until fragment manager has fragment
        boolean isIdle = fragment != null && fragment.isVisible();

        return super.handleIsIdle(isIdle);
    }
}

File: test/src/main/java/com/tunjid/androidx/test/idlingresources/FragmentVisibleIdlingResource.kt
package com.tunjid.androidx.test.idlingresources

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager

/**
 * Idling resource that waits for a fragment to be visible.
 *
 *
 * Created by tj.dahunsi on 4/29/17.
 */
class FragmentVisibleIdlingResource(
    fragmentManager: FragmentManager,
    private val fragmentTag: String
) : BaseFragmentIdlingResource(fragmentManager, fragmentTag) {

    override fun onFragmentResumed(fm: FragmentManager, f: Fragment) {
        if (f.tag == fragmentTag) idle = true
    }
}
